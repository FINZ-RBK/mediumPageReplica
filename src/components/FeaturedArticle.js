import React from "react";
import 'bootstrap/dist/css/bootstrap.css';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Image from 'react-bootstrap/Image';
import Octicon, { Star } from '@primer/octicons-react'

class FeaturedArticle extends React.Component {
    constructor(props) {
        super(props);
        this.article = props.article.article;
        this.author = props.article.articleAuthor;
    }
    render() {
        return (
            <Container >
                <Row>
                    <Col md={{ span: 4, order: 12 }} sm={{ span: 12, order: 0 }} xs={{ span: 12, order: 0 }}>
                        <Row style={{ marginLeft: 30 + "px" }}  >
                            <h3>Featured stories</h3>
                        </Row>
                        <Row style={{ marginLeft: 30 + "px", color: "#999", fontSize: 12 + "px" }}>
                            <h6>Today’s best stories, picked by our editors.</h6>
                        </Row>
                    </Col>
                    <Col md={{ span: 8, order: 0 }} sm={{ span: 12, order: 12 }} xs={{ span: 12, order: 12 }}>
                        <Row>
                            <Image src={(this.article.pic) ? this.article.pic : ""} width="100%"></Image>
                        </Row>
                        <Row style={{ marginTop: 11 + "px" }}>
                            <h1 style={{ fontSize: 3 + "vw" }}>{(this.article.title) ? this.article.title : ""}</h1>
                        </Row>
                        <Row>
                            <h6 style={{ color: "#999" }}>{(this.article.subTitle) ? this.article.subTitle : ""}</h6>
                        </Row>
                        <Row>
                            <Col md={{ span: 1, order: 0 }} style={{ padding: 0 }}>
                                <Image src={(this.author.pic) ? this.author.pic : ""} roundedCircle height="40px" width="40px"></Image>
                            </Col>
                            <Col md={{ span: 11, order: 11 }}>
                                <Row>
                                    <p style={{ fontSize: 12 + "px", marginBottom: 0 }}> {(this.author.name) ? this.author.name : ""} in Human Parts</p>
                                </Row>
                                <Row>
                                    <p style={{ fontSize: 12 + "px" }}> Nov 14 · {(this.article.readingTime) ? this.article.readingTime : ""} min read <Octicon icon={Star} size="small" verticalAlign="middle"></Octicon></p>
                                </Row>
                            </Col>
                        </Row>
                    </Col>
                </Row>
            </Container>
        );
    }
}
export default FeaturedArticle;