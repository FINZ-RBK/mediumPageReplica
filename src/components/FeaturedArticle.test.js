import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import FeaturedArticle from './FeaturedArticle';
import { exportAllDeclaration, exportSpecifier } from '@babel/types';

it('Featured Article Comonent renders without crashing', () => {
    var div = document.createElement('div');
    ReactDOM.render(<FeaturedArticle />, div);
    ReactDOM.unmountComponentAtNode(div);
});

describe('Featured Article Testing Suite', () => {
    it('snapshot renders', () => {
        var compoent = renderer.create(<FeaturedArticle headerTxt={"Featured stories"} />);
        var compJson = compoent.toJSON();
        expect(compJson).toMatchSnapshot();
    });
});




