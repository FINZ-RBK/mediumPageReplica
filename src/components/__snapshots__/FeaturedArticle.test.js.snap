// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Featured Article Testing Suite snapshot renders 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-4 col-sm-12 col-12 order-md-12 order-sm-0 order-0"
    >
      <div
        className="row"
      >
        <h3>
          Featured stories
        </h3>
      </div>
      <div
        className="row"
      >
        <h6>
          Today’s best stories, picked by our editors.
        </h6>
      </div>
    </div>
    <div
      className="col-md-8 col-sm-12 col-12 order-md-0 order-sm-12 order-12"
    >
      <div
        className="row"
      >
        <img
          className="img-fluid"
          src="https://miro.medium.com/max/3000/1*bVWy9X9vq8EYjt5ysP62gg.jpeg"
        />
      </div>
      <div
        className="row"
      >
        <h1
          style={
            Object {
              "fontSize": "3vw",
            }
          }
        >
          The Healthy Way to Write About Pain
        </h1>
      </div>
      <div
        className="row"
      >
        <h6>
          Ask the questions you were too busy, or scared, to consider in the moment
        </h6>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-1 order-md-0"
        >
          <img
            className="rounded-circle"
            height="40px"
            src="https://miro.medium.com/fit/c/80/80/1*nllGYSR4t-ORDtSOiHJacg.jpeg"
            width="40px"
          />
        </div>
        <div
          className="col-md-11 order-md-11"
        >
          <div
            className="row"
          >
            <p
              style={
                Object {
                  "fontSize": "14px",
                }
              }
            >
               Eileen Pollack in Human Parts
            </p>
          </div>
          <div
            className="row"
          >
            <p
              style={
                Object {
                  "fontSize": "14px",
                }
              }
            >
               Nov 14 · 5 min read 
              <svg
                aria-hidden="true"
                className="octicon"
                height={16}
                role="img"
                style={
                  Object {
                    "display": "inline-block",
                    "fill": "currentColor",
                    "userSelect": "none",
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 14 16"
                width={14}
              >
                <path
                  d="M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74L14 6z"
                  fillRule="evenodd"
                />
              </svg>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
